org.scala-ide.sdt.core/src/org/scalaide/ui/internal/preferences/ImplicitsPreferencePage.scala

```diff
@@ -13,46 +13,33 @@ import org.eclipse.jdt.internal.ui.preferences.PreferencesMessages
 import org.eclipse.swt.SWT
 import org.eclipse.swt.events.SelectionEvent
 import org.eclipse.ui.dialogs.PreferencesUtil
+import org.scalaide.util.eclipse.SWTUtils
 
-class ImplicitsPreferencePage extends FieldEditorPreferencePage with IWorkbenchPreferencePage {
+class ImplicitsPreferencePage extends BasicFieldEditorPreferencePage("Set the highlighting for implicit conversions and implicit parameters") {
   import ImplicitsPreferencePage._
-  import org.scalaide.util.eclipse.SWTUtils.fnToSelectionAdapter
-
-  setPreferenceStore(IScalaPlugin().getPreferenceStore)
-  setDescription("""
-Set the highlighting for implicit conversions and implicit parameters.
-  """)
 
   override def createContents(parent: Composite): Control = {
     val control = super.createContents(parent).asInstanceOf[Composite]
-    val link = new Link(control, SWT.NONE)
-    link.setText("""More options for highlighting for implicit conversions on the <a href="org.eclipse.ui.editors.preferencePages.Annotations">Text Editors/Annotations</a> preference page.""")
-    link.addSelectionListener { e: SelectionEvent =>
-      PreferencesUtil.createPreferenceDialogOn(parent.getShell, e.text, null, null)
-    }
-
+    SWTUtils.mkLinkToAnnotationsPref(parent)(a => s"More options for highlighting for implicit conversions on the $a preference page.")
     control
   }
 
   override def createFieldEditors() {
-    addField(new BooleanFieldEditor(P_ACTIVE, "Enabled", getFieldEditorParent))
-    addField(new BooleanFieldEditor(P_BOLD, "Bold", getFieldEditorParent))
-    addField(new BooleanFieldEditor(P_ITALIC, "Italic", getFieldEditorParent))
-    addField(new BooleanFieldEditor(P_CONVERSIONS_ONLY, "Only highlight implicit conversions", getFieldEditorParent))
-    addField(new BooleanFieldEditor(P_FIRST_LINE_ONLY, "Only highlight the first line in an implicit conversion", getFieldEditorParent))
+    addBooleanFieldEditors(
+      PActive -> "Enabled",
+      PBold -> "Bold",
+      PItalic -> "Italic",
+      PConversionsOnly -> "Only highlight implicit conversions",
+      PFirstLineOnly -> "Only highlight the first line in an implicit conversion")
   }
-
-  def init(workbench: IWorkbench) {}
-
 }
 
 object ImplicitsPreferencePage {
-  val BASE = "scala.tools.eclipse.ui.preferences.implicit."
-  val P_ACTIVE = BASE + "enabled"
-  val P_BOLD = BASE + "text.bold"
-  val P_ITALIC = BASE + "text.italic"
-  val P_CONVERSIONS_ONLY = BASE + "conversions.only"
-  val P_FIRST_LINE_ONLY  = BASE + "firstline.only"
+  val PActive = "scala.tools.eclipse.ui.preferences.implicit.enabled"
```

Comment:
I'll happily rename not only these, but lot's of additional constants as soon as we got a working renaming feature in the GUI.